lib/typescript/render/services/decodeURIComponentMessage.ts

```diff
@@ -0,0 +1,15 @@
+export const decodeURIComponentMessage = (messageContent: string, contentStart: string, contentEnd: string) => {
+  const enCodedMessageStartIndex = messageContent.search(contentStart);
+  const enCodedMessageStartLength = contentStart.length;
+
+  const enCodedMessageEndIndex = messageContent.search(contentEnd);
+
+  const enCodedMessage = messageContent.substring(
+    enCodedMessageStartIndex + enCodedMessageStartLength,
+    enCodedMessageEndIndex
+  );
+  const formattedEnCodedMessage = enCodedMessage.split('+').join(' ');
+  const decodedMessage = decodeURIComponent(formattedEnCodedMessage);
+
+  return decodedMessage;
+};
```

Comment:
abstracted the function that decodes URIComponentMessage in `Services`, as we were using it repeatedly for Twilio